lib/api/api_auth_webauthn.go

```diff
@@ -436,19 +407,27 @@ func (s *webauthnService) deleteOldStates() {
 	}
 }
 
-func newVolState() *WebauthnVolatileState {
-	s := WebauthnVolatileState{}
-	s.init()
+func newVolState() *apiproto.WebauthnVolatileState {
+	s := apiproto.WebauthnVolatileState{}
+	initVolatileState(&s)
 	return &s
 }
 
-func (s *WebauthnVolatileState) init() {
+func initVolatileState(s *apiproto.WebauthnVolatileState) {
 	if s.Credentials == nil {
-		s.Credentials = make(map[string]WebauthnCredentialVolatileState, 1)
+		s.Credentials = make(map[string]*apiproto.WebauthnCredentialVolatileState, 1)
 	}
 }
```

Comment:
The change to defining these structs in `proto` meant this couldn't be a method anymore because the struct definition is in another package. I couldn't seem to move the method to that package either. Am I missing some way to make that work?

Reply:
Frankly, I'd try to get rid of this method altogether if possible.  It can be integrated into `newVolState()`.  The one other usage where it's called in `loadVolatileStateRLocked()` could be inlined as well, but I doubt that `proto.Unmarshal()` will ever leave the map member uninitialized?  Though I couldn't find a reference for that behavior yet.

Reply:
Good point, done in f6269669b6ee6ed0f1a58c478013451b4ac47da6. I tested what happens if I set `.Credentials = nil` just before marshaling, and it looks like protobuf panics in that case. So yeah, it should be extremely unlikely (particularly well-formed data corruption) that protobuf unmarshals a `nil` map here.